br/pkg/lightning/backend/external/writer.go

```diff
@@ -181,27 +187,39 @@ func (b *WriterBuilder) Build(
 	}
 	p := membuf.NewPool(membuf.WithBlockNum(0), membuf.WithBlockSize(b.blockSize))
 	ret := &Writer{
-		rc: &rangePropertiesCollector{
-			props:        make([]*rangeProperty, 0, 1024),
-			currProp:     &rangeProperty{},
-			propSizeDist: b.propSizeDist,
-			propKeysDist: b.propKeysDist,
-		},
 		memSizeLimit:   b.memSizeLimit,
 		store:          store,
-		kvBuffer:       p.NewBuffer(membuf.WithMemoryLimit(b.memSizeLimit)),
 		currentSeq:     0,
+		asyncFlushCh:   make(chan *asyncFlushTask),
 		filenamePrefix: filenamePrefix,
 		keyAdapter:     keyAdapter,
 		writerID:       writerID,
-		kvStore:        nil,
 		onClose:        b.onClose,
 		closed:         false,
 		multiFileStats: make([]MultipleFilesStat, 1),
 		fileMinKeys:    make([]tidbkv.Key, 0, multiFileStatNum),
 		fileMaxKeys:    make([]tidbkv.Key, 0, multiFileStatNum),
 	}
+	ret.doubleKVBuffer[0] = p.NewBuffer(membuf.WithMemoryLimit(b.memSizeLimit))
+	ret.doubleKVBuffer[1] = p.NewBuffer(membuf.WithMemoryLimit(b.memSizeLimit))
+	ret.doubleRC[0] = &rangePropertiesCollector{
+		props:        make([]*rangeProperty, 0, 1024),
+		currProp:     &rangeProperty{},
+		propSizeDist: b.propSizeDist,
+		propKeysDist: b.propKeysDist,
+	}
+	ret.doubleRC[1] = &rangePropertiesCollector{
+		props:        make([]*rangeProperty, 0, 1024),
+		currProp:     &rangeProperty{},
+		propSizeDist: b.propSizeDist,
+		propKeysDist: b.propKeysDist,
+	}
+	ret.kvBuffer = ret.doubleKVBuffer[0]
+	ret.rc = ret.doubleRC[0]
+
 	ret.multiFileStats[0].Filenames = make([][2]string, 0, multiFileStatNum)
+	ret.wg.Add(1)
+	go ret.asyncFlush(ret.asyncFlushCh)
```

Comment:
How to cancel asyncFlush goroutine fast?

Reply:
I will add an unit test to verify it

Reply:
https://github.com/pingcap/tidb/pull/48873/commits/d4f05b40b69f820d7ef995f0548c8b3075b557bc

`asyncFlush` function will be canceled when `asyncFlushCh` is closed by `w.Close`